### PR TITLE
[karma-discord-roles] update karma-discord-roles strategy

### PR DESCRIPTION
**Pull request by Arthh**
Changes proposed in this pull request:
- Use the Snapshot block to get the timestamp.
- Accept addresses and Snapshot block timestamp in the API URL.
- Set the score = 0, if the address doesn't have the criteria. 

image

### CODE REVIEW

File: src/strategies/erc20-balance-of/index.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { formatUnits } from '@ethersproject/units';
 import { Multicaller } from '../../utils';
 
 export const author = 'bonustrack';
-export const version = '0.1.1';
+export const version = '0.1.2';
```
Comment (ChaituVR) [inline src/strategies/erc20-balance-of/index.ts:6]:
```suggestion
export const version = '0.1.1';
```

File: src/strategies/karma-discord-roles/index.ts [hunk 2]
```diff
@@ -20,27 +31,39 @@ export async function strategy(
...
+  addresses.forEach((address: string) => {
+    const userExists = delegates.find(
+      (delegate) => delegate.publicAddress === address.toLowerCase()
+    );
     const checksumAddress = getAddress(address);
-    votingPower[checksumAddress] = 1;
+    votingPower[checksumAddress] = !!userExists ? 1 : 0;
   });
 
+  console.log(votingPower);
```
Comment (ChaituVR) [inline src/strategies/karma-discord-roles/index.ts:67]:
```suggestion
```

File: src/strategies/karma-discord-roles/index.ts [hunk 3]
```diff
@@ -20,27 +31,39 @@ export async function strategy(
...
     method: 'GET',
     headers: {
       Accept: 'application/json',
       'Content-Type': 'application/json'
     }
-  });
+  };
+
+  const response = await fetch(`${KARMA_API}?${queryParams}`, requestOptions);
+  console.log(`${KARMA_API}?${queryParams}`);
```
Comment (ChaituVR) [inline src/strategies/karma-discord-roles/index.ts:52]:
```suggestion
```

File: src/strategies/erc20-balance-of/index.ts (hunk 1 above)
Comment (Arthh) [reply src/strategies/erc20-balance-of/index.ts]: sorry for the mistake, changed the wrong file.
already pushed the fix

File: src/strategies/karma-discord-roles/index.ts (hunk 3 above)
Comment (Arthh) [reply src/strategies/karma-discord-roles/index.ts]: fixed too.

**Review by ChaituVR — APPROVED** [review]

**Comment by ChaituVR** [comment]
hello @Arthh Test cases are failing. can take a look?

**Comment by Arthh** [comment]
> hello @Arthh Test cases are failing. can take a look?

Hello @ChaituVR, I ran the tests some times again and all passed 
image
image
image
image

**Comment by Arthh** [comment]
> Untitled It fails here somehow 🤔

**Comment by Arthh** [comment]
Hey @ChaituVR, I ran the tests over and over again locally and it pass.
the code always returns the addresses from params, with a score of 1 or 0.
image

**Comment by ChaituVR** [comment]
image
Just tried in my local and same issue. try passing addresses that have voting power with the latest snapshot

But anyway i will merge it. :)

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**